packages/common/src/HdWallet.ts

```diff
@@ -0,0 +1,113 @@
+import * as wasm from '@concordium/rust-bindings';
+import { mnemonicToSeedSync, validateMnemonic } from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+import { Buffer } from 'buffer/';
+import { AttributesKeys, Network } from './types';
+import { isHex } from './util';
+
+/**
+ * Class for Hierarchical Deterministic key derivation for Concordium identities and accounts.
+ */
+export class ConcordiumHdWallet {
```

Comment:
Add unit tests 1-1 with the tests we have in https://github.com/Concordium/concordium-base/blob/main/rust-src/key_derivation/src/lib.rs to ensure that all variables are passed along correctly.